types/stylelint/index.d.ts

```diff
@@ -620,34 +725,68 @@ declare namespace stylelint {
 				ignoreAtRules: OneOrMany<StringOrRegex>;
 				ignoreRules: OneOrMany<StringOrRegex>;
 				ignorePseudoClasses: OneOrMany<StringOrRegex>;
-			}
+			},
+			ExpectedMessage<[depth: number]>
+		>;
+		'media-feature-name-allowed-list': CoreRule<
+			OneOrMany<StringOrRegex>,
+			{},
+			RejectedMessage<[name: string]>
+		>;
+		'media-feature-name-disallowed-list': CoreRule<
+			OneOrMany<StringOrRegex>,
+			{},
+			RejectedMessage<[name: string]>
 		>;
-		'media-feature-name-allowed-list': CoreRule<OneOrMany<StringOrRegex>>;
-		'media-feature-name-disallowed-list': CoreRule<OneOrMany<StringOrRegex>>;
 		'media-feature-name-no-unknown': CoreRule<
 			true,
-			{ ignoreMediaFeatureNames: OneOrMany<StringOrRegex> }
+			{ ignoreMediaFeatureNames: OneOrMany<StringOrRegex> },
+			RejectedMessage<[name: string]>
 		>;
 		'media-feature-name-no-vendor-prefix': CoreRule<true>;
-		'media-feature-name-unit-allowed-list': CoreRule<Record<string, OneOrMany<string>>>;
-		'media-feature-name-value-allowed-list': CoreRule<Record<string, OneOrMany<StringOrRegex>>>;
-		'media-feature-name-value-no-unknown': CoreRule<true>;
-		'media-feature-range-notation': CoreRule<'prefix' | 'context'>;
-		'media-query-no-invalid': CoreRule<true, { ignoreFunctions: OneOrMany<StringOrRegex> }>;
+		'media-feature-name-unit-allowed-list': CoreRule<
+			Record<string, OneOrMany<string>>,
+			{},
+			RejectedMessage<[unit: string, name: string]>
```

Comment:
note

should be `[name: string, unit: string]` but that would a breaking change